axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/axon/EventBuffer.java

```diff
@@ -147,36 +147,51 @@ public void blacklist(TrackedEventMessage<?> trackedEventMessage) {
 
     @Override
     public boolean hasNextAvailable(int timeout, TimeUnit timeUnit) {
+        checkExceptionState();
         long deadline = System.currentTimeMillis() + timeUnit.toMillis(timeout);
         try {
             do {
                 long waitTime = deadline - System.currentTimeMillis();
                 waitForData(waitTime);
-            } while (peekEvent == null && System.currentTimeMillis() < deadline && !eventStream.hasNext());
+            } while (peekEvent == null && System.currentTimeMillis() < deadline);
```

Comment:
Instead of checking a field (which is set in another method), I suggest calling peek(), so that the knowledge of that field becomes more "local". To prevent the creation of Optional instances, we could create a private "peekNullable" (or any better name than that) method, which just returns null. 
That would also allow the `do ... while` to be refactored to a `while`, which is generally considered cleaner:
```
while (peekNullable() == null && System.currentTimeMillis() < deadline) {
... wait for data
}
return peekNullable() != null;
```

Reply:
Sweet, that's the kinda of clean up I was trying to find. Agreeing `peekNullable()` doesn't sound to fancy though...will give this a think on how to rename it to something better suited.